src/lib/utility.js

```diff
@@ -356,10 +356,10 @@ const getKafkaConfig = (flow, functionality, action) => {
  *
  * @returns {object} - Returns newly created general topicConfig
  */
-const createGeneralTopicConf = (functionality, action, partition = null, opaqueKey = null) => {
+const createGeneralTopicConf = (functionality, action, partition = 0, opaqueKey = null, key = null) => {
```

Comment:
I would put the key before the opaqueKey. The reason for that is that we hardly ever (I don't think we actually do use it) use it, and also if you make the key before it, it will save you specifying "null" for the opaqueKey.

Aka always arrange your function input params by the most used to the least used....therefore you don't have to specify the inputs for the least used params. <-- this means you need to change lines 131 to only include the key.

In fact I would specify the function instead as follows:

```JAVASCRIPT
const createGeneralTopicConf =  (functionality, action, key = null, partition = null) => {}
```

That way you just specify 3 params for 99% of the calls instead of having to specify ALL of them everytime.

Let me know if you do not agree.